cde-core/test-js/legacy/cdf-dd-components-selectors-spec.js

```diff
@@ -52,4 +52,38 @@ describe("CDF-DD-COMPONENTS-SELECTORS-TESTS", function() {
       expect(htmlObjectRenderer.getLabel(htmlObjectData, "${h:notInData}")).toEqual("notInData");
     });
   });
+
+  describe("ListenersRenderer #", function() {
+
+      var renderer;
+
+      beforeEach(function() {
+        renderer = new ListenersRenderer(tableManager);
+      });
+
+
+      var createProperty = function(val) {
+        return {
+          properties: [ {value: val} ]
+        };
+      };
+
+      it("getData() returns well-formed JSON object", function() {
+        var data = [ createProperty('a'), createProperty('b') ];
+        var producer = {
+          getParameters: function() {
+            return data;
+          }
+        };
+        spyOn(Panel, 'getPanel').and.returnValue(producer);
+
+        renderer.value = '["plain", "${p:parameter}"]';
```

Comment:
please test for JSON.parse SyntaxErrors:
**renderer.value = "['plain', '${p:parameter}']";**